### PR TITLE
Match DoBasicAI

### PR DESCRIPTION
**Pull request by JoshDuMan**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.02%) on `us`.

File: src/23680.c [hunk 1]
```diff
@@ -285,4 +365,84 @@ void func_8004A3E8(Evt* script, NpcAISettings* npcAISettings, EnemyTerritoryThin
...
 
-INCLUDE_ASM(s32, "23680", DoBasicAI, Evt* script, s32 isInitialCall);
+ApiStatus DoBasicAI(Evt* script, s32 isInitialCall) {
+    Enemy* enemy = script->owner1.enemy;
+    Npc* npc = get_npc_unsafe(enemy->npcID);
+    Bytecode* args = script->ptrReadPos;
+    EnemyTerritoryThing territory;
+    EnemyTerritoryThing* pTerritory = &territory;
+    NpcAISettings* aiSettings = (NpcAISettings*) evt_get_variable(script, *args++);
+
```
Comment (nanaian) [inline src/23680.c:375]: No need for extra empty line

File: src/23680.c [hunk 2]
```diff
@@ -199,7 +200,45 @@ INCLUDE_ASM(s32, "23680", func_800496B8);
...
+    if (aiSettings->unk_14 >= 0 && func_800490B4(territory, enemy, aiSettings->chaseRadius, aiSettings->unk_28.f, 0)) {
+        x = npc->pos.x;
+        y = npc->pos.y;
+        z = npc->pos.z;
+        yaw = atan2(npc->pos.x, npc->pos.z, gPlayerStatusPtr->position.x, gPlayerStatusPtr->position.z);
+        if (!npc_test_move_simple_with_slipping(npc->collisionChannel, &x, &y, &z, aiSettings->chaseSpeed, yaw, npc->collisionHeight, npc->collisionRadius)) {
+            npc->yaw = yaw;
+            ai_enemy_play_sound(npc, 0x2F4, 0x200000);
+            fx_emote(0, npc, 0, (f32) npc->collisionHeight, 1.0f, 2.0f, -20.0f, 0xF, &sp34);
+            if ((enemy->npcSettings->unk_2A & 1) != 0) {
```
Comment (ethteck) [inline src/23680.c:219]: no != 0 needed

File: src/23680.c [hunk 3]
```diff
@@ -199,7 +200,45 @@ INCLUDE_ASM(s32, "23680", func_800496B8);
...
+
+    if (aiSettings->unk_14 >= 0 && func_800490B4(territory, enemy, aiSettings->chaseRadius, aiSettings->unk_28.f, 0)) {
+        x = npc->pos.x;
+        y = npc->pos.y;
+        z = npc->pos.z;
+        yaw = atan2(npc->pos.x, npc->pos.z, gPlayerStatusPtr->position.x, gPlayerStatusPtr->position.z);
+        if (!npc_test_move_simple_with_slipping(npc->collisionChannel, &x, &y, &z, aiSettings->chaseSpeed, yaw, npc->collisionHeight, npc->collisionRadius)) {
+            npc->yaw = yaw;
+            ai_enemy_play_sound(npc, 0x2F4, 0x200000);
+            fx_emote(0, npc, 0, (f32) npc->collisionHeight, 1.0f, 2.0f, -20.0f, 0xF, &sp34);
```
Comment (ethteck) [inline src/23680.c:218]: no cast needed

File: src/23680.c [hunk 4]
```diff
@@ -199,7 +200,45 @@ INCLUDE_ASM(s32, "23680", func_800496B8);
...
+            if ((enemy->npcSettings->unk_2A & 1) != 0) {
+                script->functionTemp[0] = 0xA;
+                return;
```
Comment (ethteck) [inline src/23680.c:221]: if/else with a return after instead, numbers in decimal plez

File: src/23680.c [hunk 5]
```diff
@@ -271,7 +310,48 @@ void func_80049F7C(Evt* script, NpcAISettings* npcAISettings, EnemyTerritoryThin
...
+    Enemy* enemy = script->owner1.enemy;
+    Npc* npc = get_npc_unsafe(enemy->npcID);
+    s32 sp28;
+    f32 x, y, z;
+
+    if (!func_800490B4(territory, enemy, aiSettings->chaseRadius, aiSettings->unk_28.f, 1)) {
+        fx_emote(2, npc, 0, npc->collisionHeight, 1.0f, 2.0f, -20.0f, 15, &sp28);
+        npc->currentAnim.w = enemy->animList[0];
+        npc->duration = 20;
+        script->functionTemp[0] = 0xE;
```
Comment (ethteck) [inline src/23680.c:323]: 0xE to decimal, here and below plez

File: src/23680.c [hunk 6]
```diff
@@ -285,4 +365,83 @@ void func_8004A3E8(Evt* script, NpcAISettings* npcAISettings, EnemyTerritoryThin
...
+        }
+
+        enemy->unk_B0 &= ~4;
+        enemy->flags &= ~ENEMY_FLAGS_40000000;
+    }
+
+    switch (script->functionTemp[0]) {
+        case 0x0:
+            func_800495A0(script, aiSettings, pTerritory);
+            /* fallthrough */
```
Comment (ethteck) [inline src/23680.c:415]: comments unneeded

**Review by ethteck — CHANGES_REQUESTED** [review]

**Pull request merged**